apps/meteor/app/lib/server/functions/relinquishRoomOwnerships.ts

```diff
@@ -1,30 +1,60 @@
-import { Messages, Rooms, Subscriptions, ReadReceipts } from '@rocket.chat/models';
+import { Messages, Rooms, Subscriptions, ReadReceipts, Team } from '@rocket.chat/models';
 
 import type { SubscribedRoomsForUserWithDetails } from './getRoomsWithSingleOwner';
 import { addUserRolesAsync } from '../../../../server/lib/roles/addUserRoles';
+import { eraseTeam } from '../../../api/server/lib/eraseTeam';
 import { FileUpload } from '../../../file-upload/server';
 import { notifyOnSubscriptionChanged } from '../lib/notifyListener';
 
-const bulkRoomCleanUp = async (rids: string[]): Promise<unknown> => {
+const bulkRoomCleanUp = async (rids: string[], userId?: string) => {
 	// no bulk deletion for files
 	await Promise.all(rids.map((rid) => FileUpload.removeFilesByRoomId(rid)));
 
-	return Promise.all([
+	const rooms = await Rooms.findByIds(rids).toArray();
+
+	const teamsToRemove = rooms.filter((room) => room.teamMain);
+	const teamIds = teamsToRemove.map((room) => room.teamId).filter((teamId) => teamId !== undefined);
+	const teamPromises: Promise<void>[] = [];
+
+	teamIds.forEach((teamId) => {
+		const promise = new Promise<void>(async (resolve) => {
+			const team = await Team.findOneById(teamId);
+
+			if (!team) {
+				throw new Error('error-team-not-found');
+			}
+
+			if (!userId) {
+				throw new Error('error-user-not-found');
+			}
+
+			await eraseTeam(userId, team, []);
+
+			resolve();
+		});
+
+		teamPromises.push(promise);
+	});
+
+	await Promise.all([
 		Subscriptions.removeByRoomIds(rids, {
 			async onTrash(doc) {
 				void notifyOnSubscriptionChanged(doc, 'removed');
 			},
 		}),
 		Messages.removeByRoomIds(rids),
 		ReadReceipts.removeByRoomIds(rids),
-		Rooms.removeByIds(rids),
+		...teamPromises,
 	]);
+
+	await Rooms.removeByIds(rids);
 };
 
 export const relinquishRoomOwnerships = async function (
 	userId: string,
 	subscribedRooms: SubscribedRoomsForUserWithDetails[],
 	removeDirectMessages = true,
+	deletedBy?: string,
```

Comment:
Could we change this to deleteByUserId or set type to IUser['_id'] to make it more clear.